src/Analyzers/CSharp/CodeFixes/UseImplicitOrExplicitType/UseExplicitTypeCodeFixProvider.cs

```diff
@@ -63,63 +62,116 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
-                typeSyntax = forEach.Type;
+                await HandleForEachStatementAsync(document, editor, forEach, cancellationToken).ConfigureAwait(false);
             }
             else if (declarationContext is DeclarationExpressionSyntax declarationExpression)
             {
-                typeSyntax = declarationExpression.Type;
-                if (declarationExpression.Designation.IsKind(SyntaxKind.ParenthesizedVariableDesignation, out ParenthesizedVariableDesignationSyntax variableDesignation))
-                {
-                    parensDesignation = variableDesignation;
-                }
+                await HandleDeclarationExpressionAsync(document, editor, declarationExpression, cancellationToken).ConfigureAwait(false);
             }
             else
             {
-                throw ExceptionUtilities.UnexpectedValue(declarationContext.Kind());
+                throw ExceptionUtilities.UnexpectedValue(declarationContext?.Kind());
             }
+        }
+
+        private static async Task HandleDeclarationExpressionAsync(Document document, SyntaxEditor editor, DeclarationExpressionSyntax declarationExpression, CancellationToken cancellationToken)
+        {
+            var typeSyntax = declarationExpression.Type;
+            typeSyntax = typeSyntax.StripRefIfNeeded();
+
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
 
-            if (parensDesignation is null)
+            if (declarationExpression.Designation.IsKind(SyntaxKind.ParenthesizedVariableDesignation, out ParenthesizedVariableDesignationSyntax? variableDesignation))
             {
-                typeSyntax = typeSyntax.StripRefIfNeeded();
+                RoslynDebug.AssertNotNull(typeSyntax.Parent);
 
-                // We're going to be passed through the simplifier.  Tell it to not just convert this back to var (as
-                // that would defeat the purpose of this refactoring entirely).
-                var newTypeSyntax =
-                    semanticModel.GetTypeInfo(typeSyntax, cancellationToken).ConvertedType
-                                 .GenerateTypeSyntax(allowVar: false)
-                                 .WithTriviaFrom(typeSyntax);
+                var tupleTypeSymbol = semanticModel.GetTypeInfo(typeSyntax.Parent, cancellationToken).ConvertedType;
+                RoslynDebug.AssertNotNull(tupleTypeSymbol);
+
+                var leadingTrivia = declarationExpression.GetLeadingTrivia()
+                    .Concat(variableDesignation.GetAllPrecedingTriviaToPreviousToken().Where(t => !t.IsWhitespace()).Select(t => t.WithoutAnnotations(SyntaxAnnotation.ElasticAnnotation)));
 
-                Debug.Assert(!newTypeSyntax.ContainsDiagnostics, "Explicit type replacement likely introduced an error in code");
+                var tupleDeclaration = GenerateTupleDeclaration(tupleTypeSymbol, variableDesignation).WithLeadingTrivia(leadingTrivia);
```

Comment:
why did trivia handling change?

Reply:
hmm... good question. I wonder if this was just a bad merge somewhere when updating. I don't remember an explicit reason for this, hopefully there's a test that will fail. If not I'll try to add one and change back to `WithTriviaFrom`

Reply:
did you end up doing this?

Reply:
I did not... by just reverting it fails [OnNestedDeconstructionVarWithTrivia](https://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp.CodeStyle.UnitTests/UseExplicitTypeTests.cs,a4c88d94f8eedc81) need to investigate why

Reply:
Ah, it's just hard to follow in the diff. This correlates to lines [114](https://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp.Features/UseExplicitTypeCodeFixProvider.cs,114) thru lines [118](https://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp.Features/UseExplicitTypeCodeFixProvider.cs,118) in the previous. The only changes is that `parensDesignation` was changed to `variableDesignation` and `node` is now `declarationExpression`.

Reply:
yeah, diff is super confusing.  i trust you on this :)